atheris/instrument_bytecode.py

```diff
@@ -0,0 +1,724 @@
+            tuple(self._names),
+            get_lnotab(self._code, listing)
+        )
+    
+    def _generate_loc_invocation(self, lineno, offset):
+        """
+        Builds the bytecode that calls atheris._loc()
+        """
+        to_insert = []
+        start_offset = offset
+        name_module = self._get_name(TARGET_MODULE)
+        name_cov = self._get_name(COVERAGE_FUNCTION)
+        
+        to_insert.append(Instruction(lineno, offset, dis.opmap["LOAD_GLOBAL"], name_module))
+        offset += to_insert[-1].get_size()
+        to_insert.append(Instruction(lineno, offset, dis.opmap["LOAD_ATTR"], name_cov))
+        offset += to_insert[-1].get_size()
+        to_insert.append(Instruction(lineno, offset, dis.opmap["LOAD_CONST"], self._get_counter()))   
+        offset += to_insert[-1].get_size()
+        to_insert.append(Instruction(lineno, offset, dis.opmap["CALL_FUNCTION"], 1))
+        offset += to_insert[-1].get_size()
+        to_insert.append(Instruction(lineno, offset, dis.opmap["POP_TOP"]))
+        offset += to_insert[-1].get_size()
+        
+        return offset - start_offset, to_insert
+        
+    def _generate_cmp_invocation(self, op, lineno, offset):
+        """
+        Builds the bytecode that calls atheris._cmp().
+        Only call this if the two objects being compared are non-constants.
+        """
+        to_insert = []
+        start_offset = offset
+        name_module = self._get_name(TARGET_MODULE)
+        name_cmp = self._get_name(COMPARE_FUNCTION)
+        const_op = self._get_const(op)
+        const_pc = self._get_pc()
+        const_False = self._get_const(False)
+        
+        to_insert.append(Instruction(lineno, offset, dis.opmap["LOAD_GLOBAL"], name_module))
+        offset += to_insert[-1].get_size()
+        to_insert.append(Instruction(lineno, offset, dis.opmap["LOAD_ATTR"], name_cmp))
+        offset += to_insert[-1].get_size()
+        to_insert.append(Instruction(lineno, offset, dis.opmap["ROT_THREE"]))
+        offset += to_insert[-1].get_size()
+        to_insert.append(Instruction(lineno, offset, dis.opmap["LOAD_CONST"], const_op))
+        offset += to_insert[-1].get_size()
+        to_insert.append(Instruction(lineno, offset, dis.opmap["LOAD_CONST"], const_pc))
+        offset += to_insert[-1].get_size()
+        to_insert.append(Instruction(lineno, offset, dis.opmap["LOAD_CONST"], const_False))
+        offset += to_insert[-1].get_size()
+        to_insert.append(Instruction(lineno, offset, dis.opmap["CALL_FUNCTION"], 5))
+        offset += to_insert[-1].get_size()
+        
+        return offset - start_offset, to_insert
+        
+    def _generate_const_cmp_invocation(self, op, lineno, offset, switch):
+        """
+        Builds the bytecode that calls atheris._cmp().
+        Only call this if one of the objects being compared is a constant
+        coming from co_consts.
+        If `switch` is true the constant is the second argument and needs
```

Comment:
Ah. This explains the "only left is ever const" in my other comment. Yeah, please document over there too.